test/gtest/uct/uct_test.cc

```diff
@@ -80,14 +89,43 @@ std::vector<const resource*> uct_test::enum_resources(const std::string& tl_name
 
             for (unsigned j = 0; j < num_tl_resources; ++j) {
                 resource rsc;
-                rsc.md_name    = md_resources[i].md_name,
-                rsc.local_cpus = md_attr.local_cpus,
-                rsc.tl_name    = tl_resources[j].tl_name,
+                rsc.md_name    = md_resources[i].md_name;
+                rsc.local_cpus = md_attr.local_cpus;
+                rsc.tl_name    = tl_resources[j].tl_name;
                 rsc.dev_name   = tl_resources[j].dev_name;
                 rsc.dev_type   = tl_resources[j].dev_type;
                 all_resources.push_back(rsc);
             }
 
+            if (md_attr.cap.flags & UCT_MD_FLAG_SOCKADDR) {
+                struct ifaddrs *ifaddr, *ifa;
+                ucs_sock_addr_t sock_addr;
+
+                EXPECT_TRUE(getifaddrs(&ifaddr) != -1);
+                for (ifa = ifaddr; ifa != NULL; ifa = ifa->ifa_next) {
+                    sock_addr.addr = ifa->ifa_addr;
+                    if (uct_md_is_sockaddr_accessible(pd, &sock_addr, UCT_SOCKADDR_ACC_REMOTE) &&
+                       (ucs_netif_check(ifa->ifa_name))) {
+                        resource rsc;
+                        rsc.md_name    = md_resources[i].md_name;
+                        rsc.local_cpus = md_attr.local_cpus;
+                        rsc.tl_name    = "sockaddr";
+                        rsc.dev_name   = ifa->ifa_name;
+                        rsc.dev_type   = UCT_DEVICE_TYPE_LAST;
```

Comment:
why?

Reply:
if this field isn't set, valgrind complains about it when checking it in several tests.
for example - https://github.com/openucx/ucx/blob/master/test/gtest/uct/uct_p2p_test.cc#L41

Reply:
should be NET